pkg/bootstrap/types.go

```diff
@@ -73,10 +73,11 @@ type BootstrapResult struct {
 }
 
 type InfraStatus struct {
-	PlatformType         configv1.PlatformType
-	PlatformRegion       string
-	PlatformStatus       *configv1.PlatformStatus
-	ExternalControlPlane bool
```

Comment:
I felt it was best to delete `ExternalControlPlane` (and its test `pkg/platform/platform_test.go`) because it would in a way contain duplicate information that's already carried by the newly added `ControlPlaneTopology` and `InfrastructureTopology` struct fields.

I'm willing to undo this change, if required